### PR TITLE
Polyhedron_demo: Fix images issues in mesh 3 plugin

### LINKED ISSUE #1182: Issues with the Mesh_3 plugin for images
@maxGimeno 
- In `Io_image_plugin.cpp`, `struct controls` (added by 7e422d4) does not follow the naming convention. All types in CGAL should start by a capital letter. Please read the [CGAL coding convention][c]. And I mean it: you *must* read them, as any CGAL developer. Note that in CGAL Qt5 demos, it is accepted to follow the ["camelCase"][cc] naming convention for method and classes names. But anyway, in both CGAL or Qt5 convention, a type must start with a capital letter. The first author of volume planes (@bo0ts) was also not always following the conventions, and his code was also annoying to read.
[c]: http://doc.cgal.org/latest/Manual/devman_code_format.html
[cc]: https://en.wikipedia.org/wiki/CamelCase
- During the creation of the volume planes, the application cursor should be the `Qt::waitCursor`.
- One cannot pick the planes using the shift+right.
- The `bbox()` for volume plane is wrong: it seems uninitialized.
- `Scene_image_item::toolTip()` always display "Word type: ... ". That is hard-coded in the code. That makes hard to understand the popup dialog that says: "*Selected object can't be meshed because the image's word type is not supported by this plugin.*"
- For a gray image, `Vertex_buffer_helper::fill_buffer_data()` is called anyway, with `dx=dy=dz=125`.

### PR DESCRIPTION
**Pull request by maxGimeno**
This PR fixes #1182.

This is a fix for the Io_image_plugin:
- This corrects the naming style
- This fixes the Word type in the toolTip()
- This adds a waitCursor during the creation of the planes
- This adds a bbox to the volume planes
- This keeps the item from computing useless data for the gray-images display.

### CODE REVIEW

File: Polyhedron/demo/Polyhedron/Scene_image_item.cpp [hunk 1]
```diff
@@ -654,15 +654,33 @@ Scene_image_item::draw(Viewer_interface* viewer) const
   }
 }
 
+template <typename T> const char* whatType(T) { return "unknown"; }    // default
+template <> const char* whatType(float) { return "float"; }    // default
+template <> const char* whatType(double) { return "double"; }    // default
+template <> const char* whatType(char) { return "char"; }    // default
+template <> const char* whatType(boost::uint8_t) { return "uint8 "; }    // default
+template <> const char* whatType(boost::int16_t) { return "uint16"; }    // default
+template <> const char* whatType(boost::uint16_t type) { return "uint16_t"; }    // default
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_image_item.cpp]: The display for `uint8_t` and `uint16_t` miss the `_t` at the end.

And, to be consistent with the other integral types, you should replace `char` by `int8_t`.

**Comment by lrineau** [comment]
I made one minor comment, see https://github.com/CGAL/cgal/pull/1184#discussion_r67316876, otherwise everything is right. Good job!

**Comment by maxGimeno** [comment]
Fixed

File: Polyhedron/demo/Polyhedron/Scene_image_item.cpp [hunk 2]
```diff
@@ -655,14 +655,14 @@ Scene_image_item::draw(Viewer_interface* viewer) const
...
-template <> const char* whatType(boost::uint8_t) { return "uint8 "; }    // default
-template <> const char* whatType(boost::int16_t) { return "uint16"; }    // default
-template <> const char* whatType(boost::uint16_t type) { return "uint16_t"; }    // default
-template <> const char* whatType(boost::int32_t) { return "int32_t"; }    // default
-template <> const char* whatType(boost::uint32_t) { return "uint32_t"; }    // default
+template <> const char* whatType(float) { return "float"; }
+template <> const char* whatType(double) { return "double"; }
+template <> const char* whatType(char) { return "int8_t (char)"; }
+template <> const char* whatType(boost::uint8_t) { return "uint8_t"; }
+template <> const char* whatType(boost::int16_t) { return "uint16_t"; }
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/Scene_image_item.cpp]: If you want to translate the `uintN_t` and `intN_t`, you can use:
  - `unsigned char` for `uint8_t`,
  - `short` for `int16_t`,
  - `unsigned short` for `uint16_t`.
  - `int` for `int32_t`,
  - `unsigned int` for `uint32_t`.
That list of conversion is true [on a lot of platforms][*], and on all platforms supported by CGAL.

[*]: https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models

**Comment by maxGimeno** [comment]
This should be ready to merge.

**Pull request merged**